mrp_production_add_middle_stuff/wizard/wizard_addition.py

```diff
@@ -51,5 +52,7 @@ def add_product(self):
                                                 self.product_qty,
                                                 self.production_id)
         line = mppl_obj.create(values)
-        production_obj._make_production_consume_line(line)
+        move = production_obj._make_production_consume_line(line)
+        move_obj.browse(move).action_confirm()
+        self.production_id.action_assign()
```

Comment:
This line should depend on the state of the production, because you can add middle stuff when confirmed but not checked availability.